### PR TITLE
task(js-post-oop): Add requirements

### PR DESCRIPTION
**Pull request by OleksiyRudenko**
Adding requirements and hints re OOP/SOLID to address the most common issues in code submitted by the students.

[Doc render preview](https://github.com/kottans/frontend/blob/425e4508152e777379109fa3a3af2572ccbd7cae/tasks/js-post-oop.md)

### CODE REVIEW

**Comment by suchov** [comment]
@OleksiyRudenko SOLID is not an easy topic are you sure you want to put it to the requirements?
maybe just as a point to catch the interest, but not sure that it can be a requirement.
https://medium.com/webbdev/solid-4ffc018077da what do you think about adding some links like that to the extra materials or something?

File: tasks/js-post-oop.md [hunk 1]
```diff
@@ -8,13 +8,33 @@
...
+
 1. Improve your tiny JS world model you forked from
    [here](https://github.com/OleksiyRudenko/a-tiny-JS-world).
    This time use your favourite JS OOP methods to keep
    your code DRY and SOLID.
 
    You don't need forking this time, work with the repo you've
    already got. Git keeps history for you.
+   
+   You will find [these materials](https://github.com/OleksiyRudenko/a-tiny-JS-world/blob/master/README.md#learn-on-your-own) helpful.
```
Comment (OleksiyRudenko) [inline tasks/js-post-oop.md:21]:
```suggestion
   You will find materials under section **Learn on your own** [here](https://github.com/OleksiyRudenko/a-tiny-JS-world/blob/master/README.md#learn-on-your-own) helpful.
```

File: tasks/js-post-oop.md [hunk 2]
```diff
@@ -8,13 +8,33 @@
...
+- [ ] If peer classes (siblings) have same properties those must be generalized to their base (parent) class
+- [ ] Classes for species that do not have hands by natural design, do not consequently have hands or any equivalent property
+      and do not inherit such properties from any of the base/parent classes
+- [ ] Class properties that are parts of an object's presentation string are listed/specified explicitly
+      (i.e. `Object` methods `values`, `keys`, `entires` or constructions like `for (key in obj)` aren't used)
+- [ ] Object's presentation string builder is a class method, not an independent function
+- [ ] Class constructor is defined with named parameters
+- [ ] A class doesn't assign or change property owned by any other class (defined in any other class)
+- [ ] A child class shouldn't directly use an inherited property to build an object's presentation string
+- [ ] The code is OOP, SOLID and DRY compliant
```
Comment (OleksiyRudenko) [inline tasks/js-post-oop.md:34]:
```suggestion
- [ ] The code is [OOP](https://www.freecodecamp.org/news/object-oriented-programming-concepts-21bb035f7260/), SOLID ([eng](https://medium.com/@cramirez92/s-o-l-i-d-the-first-5-priciples-of-object-oriented-design-with-javascript-790f6ac9b9fa), [rus](https://medium.com/webbdev/solid-4ffc018077da)) and [DRY](https://code.tutsplus.com/tutorials/3-key-software-principles-you-must-understand--net-25161) compliant
```

**Review by OleksiyRudenko — COMMENTED** [review]
Add links to reference materials

**Comment by OleksiyRudenko** [comment]
@suchov you're right. Suggested a couple of links for reference.
Please check.

**Pull request merged**